Objects/longobject.c

```diff
@@ -4565,11 +4595,23 @@ _PyLong_Rshift(PyObject *a, size_t shiftby)
 static PyObject *
 long_lshift1(PyLongObject *a, Py_ssize_t wordshift, digit remshift)
 {
-    /* This version due to Tim Peters */
     PyLongObject *z = NULL;
     Py_ssize_t oldsize, newsize, i, j;
     twodigits accum;
 
+    if (wordshift == 0 && IS_MEDIUM_VALUE(a) &&
+        (a->ob_digit[0] & ~(PyLong_MASK >> remshift)) == 0) {
+        stwodigits sval = (stwodigits)((twodigits)medium_value(a) << remshift);
+        // bypass undefined shift operator behavior
```

Comment:
I would move this comment one line up. PEP 7 says "Comments go before the code they describe."